testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +196,282 @@ def process_test_clean(test_tag, work_dir):  # {{{
     # Clean test case
     subprocess.check_call(
         ['./clean_testcase.py', '-q', '--work_dir', work_dir, '-o', test_core,
-         '-c',  test_configuration, '-r', test_resolution, '-t', test_test],
+         '-c', test_configuration, '-r', test_resolution, '-t', test_test],
         stdout=dev_null, stderr=dev_null)
 
     print("   -- Cleaned case '{}': -o {} -c {} -r {} -t {}".format(
         test_name, test_core, test_configuration, test_resolution, test_test))
 
     dev_null.close()
+    # }}}
+
+
+def local_parallel_setup_script(
+    args, suite_tag, testcase_data, testcase_data_prereq, data_dic):
+    # {{{
+    work_dir = args.work_dir
+    verbose = args.verbose
+    suite_root = suite_tag
+    try:
+        suite_name = suite_tag.attrib['name']
+    except KeyError:
+        print("ERROR: <regression_suite> tag is missing 'name' attribute.")
+        print('Exiting...')
+        sys.exit(1)
+
+    if not os.path.exists('{}'.format(work_dir)):
+        os.makedirs('{}'.format(work_dir))
+
+    if verbose:
+        # flush existing regression suite output file
+        open(work_dir + '/manage_regression_suite.py.out', 'w').close()
+
+    regression_script_name = '{}/parallel_{}.py'.format(work_dir, suite_name)
+    local_parallel_script = open('{}'.format(regression_script_name), 'w')
+    local_parallel_code = write_regression_local_parallel_top(
+        work_dir, suite_tag, args.nodes, data_dic)
+
+
+    dev_null = open('/dev/null', 'a')
+    subprocess.check_call(['chmod',
+                           'a+x',
+                           '{}'.format(regression_script_name)],
+                           stdout=dev_null,
+                           stderr=dev_null)
+    dev_null.close()
+
+
+    setup_suite(
+        suite_root,
+        args.work_dir,
+        args.model_runtime,
+        args.config_file,
+        args.baseline_dir,
+        args.verbose,
+        args,
+        data_dic)
+
+    local_parallel_code = write_regression_local_parallel_testcase_data(
+        local_parallel_code, work_dir, testcase_data,  testcase_data_prereq, data_dic)
+
+    local_parallel_code = write_regression_local_parallel_bottom(
+        local_parallel_code)
+
+    return local_parallel_script, local_parallel_code
+    # }}}
+
+
+def get_prereq(testcase_data, testcase_data_prereq, data_dic):
+    # {{{
+    names = [i for i in testcase_data.keys() if i in testcase_data_prereq]
+    cwd = os.getcwd() + "/"
+    for i in names:
+        command = ['time', '-p', str(cwd)+str(testcase_data[i]['path'])]
```

Comment:
My previous comment got lost in other changes, but we cannot use `os.getcwd()` because this does not work with the `--work_dir` argument.  Please always test with the `--work_dir` argument and point it to a different directory (e.g. on scratch somewhere).

Reply:
updated to use the args.work_dir in lieu of os.getcwd